diff -ru Linux-PAM-1.5.2/modules/Makefile.am Linux-PAM-1.5.2-new/modules/Makefile.am
--- Linux-PAM-1.5.2/modules/Makefile.am	2021-09-03 13:59:07.000000000 +0200
+++ Linux-PAM-1.5.2-new/modules/Makefile.am	2023-06-02 16:00:20.481100281 +0200
@@ -43,50 +43,7 @@
 endif
 
 SUBDIRS := \
-	pam_access \
-	pam_debug \
-	pam_deny \
-	pam_echo \
-	pam_env \
-	pam_exec \
-	pam_faildelay \
-	pam_faillock \
-	pam_filter \
-	pam_ftp \
-	pam_group \
-	pam_issue \
-	$(MAYBE_PAM_KEYINIT) \
-	$(MAYBE_PAM_LASTLOG) \
-	pam_limits \
-	pam_listfile \
-	pam_localuser \
-	pam_loginuid \
-	pam_mail \
-	pam_mkhomedir \
-	pam_motd \
-	$(MAYBE_PAM_NAMESPACE) \
-	pam_nologin \
-	pam_permit \
-	pam_pwhistory \
-	$(MAYBE_PAM_RHOSTS) \
-	pam_rootok \
-	pam_securetty \
-	$(MAYBE_PAM_SELINUX) \
-	$(MAYBE_PAM_SEPERMIT) \
-	$(MAYBE_PAM_SETQUOTA) \
-	pam_shells \
-	pam_stress \
-	pam_succeed_if \
-	pam_time \
-	pam_timestamp \
-	$(MAYBE_PAM_TTY_AUDIT) \
-	pam_umask \
 	$(MAYBE_PAM_UNIX) \
-	$(MAYBE_PAM_USERDB) \
-	pam_usertype \
-	pam_warn \
-	pam_wheel \
-	pam_xauth \
 	#
 
 CLEANFILES = *~
diff -ru Linux-PAM-1.5.2/modules/Makefile.in Linux-PAM-1.5.2-new/modules/Makefile.in
--- Linux-PAM-1.5.2/modules/Makefile.in	2021-09-03 13:59:18.000000000 +0200
+++ Linux-PAM-1.5.2-new/modules/Makefile.in	2023-06-02 17:15:01.273484486 +0200
@@ -169,15 +169,7 @@
   done | $(am__uniquify_input)`
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = pam_access pam_debug pam_deny pam_echo pam_env pam_exec \
-	pam_faildelay pam_faillock pam_filter pam_ftp pam_group \
-	pam_issue pam_keyinit pam_lastlog pam_limits pam_listfile \
-	pam_localuser pam_loginuid pam_mail pam_mkhomedir pam_motd \
-	pam_namespace pam_nologin pam_permit pam_pwhistory pam_rhosts \
-	pam_rootok pam_securetty pam_selinux pam_sepermit pam_setquota \
-	pam_shells pam_stress pam_succeed_if pam_time pam_timestamp \
-	pam_tty_audit pam_umask pam_unix pam_userdb pam_usertype \
-	pam_warn pam_wheel pam_xauth
+DIST_SUBDIRS = pam_unix
 am__DIST_COMMON = $(srcdir)/Makefile.in
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
@@ -398,50 +390,7 @@
 @COND_BUILD_PAM_UNIX_TRUE@MAYBE_PAM_UNIX = pam_unix
 @COND_BUILD_PAM_USERDB_TRUE@MAYBE_PAM_USERDB = pam_userdb
 SUBDIRS := \
-	pam_access \
-	pam_debug \
-	pam_deny \
-	pam_echo \
-	pam_env \
-	pam_exec \
-	pam_faildelay \
-	pam_faillock \
-	pam_filter \
-	pam_ftp \
-	pam_group \
-	pam_issue \
-	$(MAYBE_PAM_KEYINIT) \
-	$(MAYBE_PAM_LASTLOG) \
-	pam_limits \
-	pam_listfile \
-	pam_localuser \
-	pam_loginuid \
-	pam_mail \
-	pam_mkhomedir \
-	pam_motd \
-	$(MAYBE_PAM_NAMESPACE) \
-	pam_nologin \
-	pam_permit \
-	pam_pwhistory \
-	$(MAYBE_PAM_RHOSTS) \
-	pam_rootok \
-	pam_securetty \
-	$(MAYBE_PAM_SELINUX) \
-	$(MAYBE_PAM_SEPERMIT) \
-	$(MAYBE_PAM_SETQUOTA) \
-	pam_shells \
-	pam_stress \
-	pam_succeed_if \
-	pam_time \
-	pam_timestamp \
-	$(MAYBE_PAM_TTY_AUDIT) \
-	pam_umask \
 	$(MAYBE_PAM_UNIX) \
-	$(MAYBE_PAM_USERDB) \
-	pam_usertype \
-	pam_warn \
-	pam_wheel \
-	pam_xauth \
 	#
 
 CLEANFILES = *~
