diff --git a/Modules/Platform/WindowsPaths.cmake b/Modules/Platform/WindowsPaths.cmake
index b9e2f17979..ab517cd4a7 100644
--- a/Modules/Platform/WindowsPaths.cmake
+++ b/Modules/Platform/WindowsPaths.cmake
@@ -71,7 +71,18 @@ endif()

 if(CMAKE_CROSSCOMPILING AND NOT CMAKE_HOST_SYSTEM_NAME MATCHES "Windows")
   # MinGW (useful when cross compiling from linux with CMAKE_FIND_ROOT_PATH set)
   # list(APPEND CMAKE_SYSTEM_PREFIX_PATH /)
+if(DEFINED ENV{NIX_CC}
+  AND IS_DIRECTORY "$ENV{NIX_CC}"
+  AND EXISTS "$ENV{NIX_CC}/nix-support/orig-libc"
+  AND EXISTS "$ENV{NIX_CC}/nix-support/orig-libc-dev")
+  file(STRINGS "$ENV{NIX_CC}/nix-support/orig-libc" _nix_cmake_libc)
+  file(STRINGS "$ENV{NIX_CC}/nix-support/orig-libc-dev" _nix_cmake_libc_dev)
+else()
+  set(_nix_cmake_libc @libc_lib@)
+  set(_nix_cmake_libc_dev @libc_dev@)
+endif()
+
 endif()

 list(APPEND CMAKE_SYSTEM_INCLUDE_PATH
